apps/web/src/pages/liquidity/index.tsx

```diff
@@ -95,26 +95,24 @@ export default function PoolListPage() {
   const isNeedFilterByQuery = useMemo(() => token0 || token1 || fee, [token0, token1, fee])
   const [showAllPositionWithQuery, setShowAllPositionWithQuery] = useState(false)
 
-  let v2PairsSection: null | JSX.Element[] = null
+  const v2PairsSection: null | JSX.Element[] = useMemo(() => {
+    if (!v2Pairs?.length) return null
 
-  if (v2Pairs?.length) {
-    v2PairsSection = v2Pairs.map((pair, index) => (
+    return v2Pairs.map((pair, index) => (
       // eslint-disable-next-line react/no-array-index-key
       <V2PairCard key={`${pair?.token0}-${pair?.token1}-${index}`} pair={pair} account={account} />
     ))
-  }
+  }, [account, v2Pairs])
```

Comment:
I don't think it's necessary to use `useMemo` on jsx. If we want to optimize the performance, let's use `React.memo` on the component itself.

Memoyil and I had a discussion earlier on this topic https://github.com/pancakeswap/pancake-frontend/pull/7921#discussion_r1335515315